### ISSUE TITLE
Adopt Flathub ET Legacy as a full member of the product release

### ISSUE THREAD

**Issue by Eonfge** [etlegacy/etlegacy#1622]
It has now been a few weeks since the release of 2.77, and since then I and some other contributors have made a 32bit flathub release:
https://flathub.org/apps/details/com.etlegacy.ETLegacy

This release for Linux has many benefits, including:
- Run 32bit ET Legacy on a 64bit machine
- Run ET Legacy in a container to protect the user's files and system
- Automatically download all required assets from Splash Damage

 It truly is a one-click install for almost all Linux systems out there.

### Getting involved?

For now, the maintenance of this package is done by volunteers like me, which means that it sometimes takes a day or two to sync the Flathub version with the official version. It's not hard work, but it does take a small review process. I would ask you as upstream, if you're interested in getting directly involved.

Things that you can do:
- When there is a new ETL release, update the main package publish it in Flathub
- Add a button on your downloads page to directly link to it

Feel free to ask, if you have any questions.

**Comment by jackeri** [comment]
Happy to see that there is a working flatpak package now. I personally have no experience with the flatpak system, and I don't know if anyone in the project team has either. Depending on how high/low maintenance it is we should maybe consider it.

**Comment by rmarquis** [comment]
No experience either, but if I may, what about the obvious solution: promoting @Eonfge to official ETL maintainer of the flatpak package? 😜

**Comment by Eonfge** [comment]
Whaha, I'll keep doing what I do best, if you can keep me posted on when there are new releases.

As for what the Flatpak does, it's a lot of scaffolding to support 32 bit, follewed by:
- Download the etlegacy zip from the main site
- Install the bins and renderer to `/app/bin/` and copy paste the `etmain` and `legacy` inside the `/app/bin/`
- Create symlinks from the `/app/extra/` directory, which is the location to download the proprietary .pk3 on installation
- Edit the desktop-file so that it's basepath is set to `/app/bin/`

Then, there is a small post-installation script that runs in `/app/extra/`:
- Download W:ET
- Run installer, keygen and game without UI
- Copy paste pk3 files 
- remove everything not pk3

When the user clicks the desktop icon, the user can run the game all without manual configuration.There is also a single-line bash script for CLI users, so that it automatically sets the base-path.

The full config file is here:
https://github.com/flathub/com.etlegacy.ETLegacy/blob/master/com.etlegacy.ETLegacy.yaml

### Possible improvements

Honestly, The only way to make this smoother, is by supporting some kind of environment variable like BASEPATH=/app/bin/, should I make a second issue for that?

_Normally, we try to compile everything from source but since this is a 32bit application, running with lots of legacy dependencies, we decided to make en exemption there._

**Comment by isRyven** [comment]
it should be enough to run etlegacy installation script in place of vet one to get all necessary assets

**Comment by Eonfge** [comment]
@isRyven 
That's something that's a bit against the design of ETLegacy. While the system can technically be made in such a way that it will just run an installation script inside your ~/home/ directory, it's preferred to have all the files pre-packaged because of the way how Flatpak works.

Some special details of Flatpaks:
- The application directory is read-only
- The application directory is imaged using [libostree](https://ostreedev.github.io/ostree/) (Git for applications)
- All updates and changes to the application are atomic:
  - Only update the files that change
  - Only complete updates when the application is safely existed

This is all great for reliability and security, but it works best when applications don't try to work against it. The thing you propose can be done and it's a possible workaround for non-free software but it's certainly not recommended.

**Comment by isRyven** [comment]
Sorry i still don't really understand why it's against the design of ETLegacy or why it's not desired, is this because of possible license infringement? 

But just to be clear, so we are on the same page, i was specificially referring to these steps, which i find completely extra:

> Then, there is a small post-installation script that runs in /app/extra/:
>    Download W:ET
>    Run installer, keygen and game without UI
>    Copy paste pk3 files
>    remove everything not pk3

I don't really see the reason of running vet installation, when it's enough to jut run etlegacy's one, which will also fetch all the game assets (pak0-2) game really needs to run, i don't know what's the deal about it, and why you need any complexities you have implemented as extra steps, unless, again, you think the etlegacy installer is violating some license (although it's quite subjective imo).

**Comment by Eonfge** [comment]
Yes, it's because of the legal limits within Flathub. Flathub is fine with providing non-free applications, but without the explicit agreement from the owner [(Steam Link for example)](https://flathub.org/apps/details/com.valvesoftware.SteamLink), they'll only allow the 'download instructions' workaround.

It's quite a nice legal loophole really, and some parties have complained about it: I can explain you how to install Wolfenstein ET in a container using UK English, so why not explain it to use using Bash English. This is also why you'll find Skype, Teams or Minecraft on Flathub... although Microsoft is ominously silent on the issue.

**Comment by rmarquis** [comment]
The workaround you describe above is *exactly* what the ET:L installer script does (so it doesn't violate any license). Unless you missed that point, I am not sure what is the issue here.

**Comment by Eonfge** [comment]
> The workaround you describe above is _exactly_ what the ET:L installer script does (so it doesn't violate any license). Unless you missed that point, I am not sure what is the issue here.

Ow, I tried that but there are two issues with that approach:
- Like most build systems on Linux, Flatpak compiles and installs without network access, I can't sneak in a script to download extra assets, even if that is desirable in this situation. All archives and files must de declared in advance, with a proper sha256 and/or the exact size.
- There is a bug in the installer, when using `--skip-license` it won't download the W:ET .pk3 files. This makes it useless in automated background installations and updates.

I know, it all sounds like Flathub is a lot of clumsy work, but in the end it does the same as your .sh script, it just happens to do these steps in a Flathub-defined order and with Flathub's security constraints. 

Fear not, I'll make it work. If you can just keep me posted on updates, that's most important

**Comment by rmarquis** [comment]
Fair enough.

To stay informed on releases, you can simply subscribe to the GitHub repo (Watch button -> Custom -> Releases). You'll get a notification and/or email when a release tag is added. Alternatively, you can follow the Discord channel, or the [official blog](https://www.etlegacy.com/blog).

**Comment by Eonfge** [comment]
Let's see how the adaption goes. If you want, I can always grant you maintainer status later.
